modules/imgcodecs/test/test_png.cpp

```diff
@@ -110,6 +110,45 @@ TEST(Imgcodecs_Png, read_color_palette_with_alpha)
     EXPECT_EQ(img.at<Vec3b>(0, 1), Vec3b(255, 0, 0));
 }
 
+// IHDR shall be first.
+// See https://github.com/opencv/opencv/issues/27295
+TEST(Imgcodecs_Png, decode_regression27295)
+{
+    vector<uchar> buff;
+    Mat src = Mat::zeros(240, 180, CV_8UC3);
+    vector<int> param;
+    EXPECT_NO_THROW(imencode(".png", src, buff, param));
+
+    Mat img;
+
+    // If first is IHDR chunk, output shall not be empty.
+    // 8 means PNG sigunature length.
```

Comment:
signature

Reply:
Thank you for your review, I fixed this and next line.
(IHDR does not expect the length of chunk data is 4.)

```diff
-    // If first is IHDR chunk, output shall not be empty.
-    // 8 means PNG sigunature length.
-    // 4 means lenght of chunk.
+    // If IHDR chunk found as the first chunk, output shall not be empty.
+    // 8 means PNG signature length.
+    // 4 means length field(uint32_t).
```